packages/extension/src/companion/companionRequest.tsx

```diff
@@ -0,0 +1,16 @@
+import request from 'graphql-request';
+import { browser } from 'webextension-polyfill-ts';
+
+const proxyRequest = {
+  apply(_, __, args) {
+    browser.runtime.sendMessage({
+      type: 'GRAPHQL_REQUEST',
+      url: args?.[0],
+      document: args?.[1],
+      variables: args?.[2],
+    });
+    return null;
+  },
+};
+
+export const companionRequest = new Proxy(request, proxyRequest);
```

Comment:
What's the advantage of using `Proxy`?
Because it seems like you must import the `graphql-request` module here although you don't really use it

Reply:
I think the main advantage is that is copies the signature so we don't have to explicitly state those, and we can simply "modify" what it actually does.

Would you rather have a function that just has the same props?

Reply:
I would favor a solution that doesn't involve importing unnecessary modules in this case